create_database.sql

```diff
@@ -0,0 +1,1732 @@
+                                        calculated_score, created_by, created_on)
+        VALUES (mapId, sampleId, CASE WHEN instrumentLastCalibrated = '' THEN NULL ELSE instrumentLastCalibrated END,
+                @formattedErrorCode, CASE WHEN dateTested = '' THEN NULL ELSE dateTested END, @formattedSampleMtbDetected,
+                @formattedSampleRifResistance,
+                CASE WHEN probeD IN ('', 'No Result', 'ERROR', 'NA', 'POSITIVE', 'NEGATIVE', 'NOT REPORTED', 'APROVED', 'PASS', 'NEG', 'POS', 'N/A', '-') THEN '' ELSE CAST(TRIM(REPLACE(REPLACE(probeD, ',', '.'), 'O', '0')) AS DECIMAL(10,1)) END,
+                CASE WHEN probeC IN ('', 'No Result', 'ERROR', 'NA', 'POSITIVE', 'NEGATIVE', 'NOT REPORTED', 'APROVED', 'PASS', 'NEG', 'POS', 'N/A', '-') THEN '' ELSE CAST(TRIM(REPLACE(REPLACE(probeC, ',', '.'), 'O', '0')) AS DECIMAL(10,1)) END,
+                CASE WHEN probeE IN ('', 'No Result', 'ERROR', 'NA', 'POSITIVE', 'NEGATIVE', 'NOT REPORTED', 'APROVED', 'PASS', 'NEG', 'POS', 'N/A', '-') THEN '' ELSE CAST(TRIM(REPLACE(REPLACE(probeE, ',', '.'), 'O', '0')) AS DECIMAL(10,1)) END,
+                CASE WHEN probeB IN ('', 'No Result', 'ERROR', 'NA', 'POSITIVE', 'NEGATIVE', 'NOT REPORTED', 'APROVED', 'PASS', 'NEG', 'POS', 'N/A', '-') THEN '' ELSE CAST(TRIM(REPLACE(REPLACE(probeB, ',', '.'), 'O', '0')) AS DECIMAL(10,1)) END,
+                CASE WHEN spcProbe IN ('', 'No Result', 'ERROR', 'NA', 'POSITIVE', 'NEGATIVE', 'NOT REPORTED', 'APROVED', 'PASS', 'NEG', 'POS', 'N/A', '-') THEN '' ELSE CAST(TRIM(REPLACE(REPLACE(spcProbe, ',', '.'), 'O', '0')) AS DECIMAL(10,1)) END,
+                CASE WHEN probeA IN ('', 'No Result', 'ERROR', 'NA', 'POSITIVE', 'NEGATIVE', 'NOT REPORTED', 'APROVED', 'PASS', 'NEG', 'POS', 'N/A', '-') THEN '' ELSE CAST(TRIM(REPLACE(REPLACE(probeA, ',', '.'), 'O', '0')) AS DECIMAL(10,1)) END,
+                CASE WHEN sampleScore > 0 THEN 'pass' ELSE 'fail' END, 1, NOW());
+    END IF;
+END ;;
+DELIMITER ;
+/*!50003 SET sql_mode              = @saved_sql_mode */ ;
+/*!50003 SET character_set_client  = @saved_cs_client */ ;
+/*!50003 SET character_set_results = @saved_cs_results */ ;
+/*!50003 SET collation_connection  = @saved_col_connection */ ;
+/*!50003 DROP PROCEDURE IF EXISTS `UpdateInstrument` */;
+/*!50003 SET @saved_cs_client      = @@character_set_client */ ;
+/*!50003 SET @saved_cs_results     = @@character_set_results */ ;
+/*!50003 SET @saved_col_connection = @@collation_connection */ ;
+/*!50003 SET character_set_client  = utf8 */ ;
+/*!50003 SET character_set_results = utf8 */ ;
+/*!50003 SET collation_connection  = utf8_general_ci */ ;
+/*!50003 SET @saved_sql_mode       = @@sql_mode */ ;
+/*!50003 SET sql_mode              = 'STRICT_TRANS_TABLES,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION' */ ;
+DELIMITER ;;
+CREATE DEFINER=`root`@`localhost` PROCEDURE `UpdateInstrument`()
+BEGIN
+	ALTER TABLE `instrument` ADD UNIQUE( `participant_id`, `instrument_serial`);
+END ;;
+DELIMITER ;
```

Comment:
Please don't create a store procedure to add this sort of thing? Rather just edit it on the table definition for instrument. The way that we currently deploy these changes is to run this entire script file on a new, clean database, and then compare this with the live database, using Redgate MySQL Compare to create a change script.

Reply:
Ok you mean not to create procedure for this change is that?

Reply:
Yes. So the way that you should bring this change, into the script is to change:
```
CREATE TABLE `instrument` (
  `instrument_id` int(11) NOT NULL AUTO_INCREMENT,
  `participant_id` int(11) NOT NULL,
  `instrument_serial` varchar(45) DEFAULT NULL,
  `instrument_installed_on` date DEFAULT NULL,
  `instrument_last_calibrated_on` date DEFAULT NULL,
  `created_by` varchar(45) DEFAULT NULL,
  `created_on` datetime DEFAULT NULL,
  `updated_by` varchar(45) DEFAULT NULL,
  `updated_on` datetime DEFAULT NULL,
  PRIMARY KEY (`instrument_id`),
  KEY `participant_id` (`participant_id`),
  CONSTRAINT `instrument_ibfk_2` FOREIGN KEY (`participant_id`) REFERENCES `participant` (`participant_id`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
to 
```
CREATE TABLE `instrument` (
  `instrument_id` int(11) NOT NULL AUTO_INCREMENT,
  `participant_id` int(11) NOT NULL,
  `instrument_serial` varchar(45) DEFAULT NULL,
  `instrument_installed_on` date DEFAULT NULL,
  `instrument_last_calibrated_on` date DEFAULT NULL,
  `created_by` varchar(45) DEFAULT NULL,
  `created_on` datetime DEFAULT NULL,
  `updated_by` varchar(45) DEFAULT NULL,
  `updated_on` datetime DEFAULT NULL,
  PRIMARY KEY (`instrument_id`),
  KEY `participant_id` (`participant_id`),
  CONSTRAINT `instrument_ibfk_2` FOREIGN KEY (`participant_id`) REFERENCES `participant` (`participant_id`),
  UNIQUE KEY `participant_id_instrument_serial` (`participant_id`, `instrument_serial`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```